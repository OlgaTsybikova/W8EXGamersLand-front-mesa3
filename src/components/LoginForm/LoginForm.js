import LoginFormContainer from "./LoginFormStyled";
import { useState } from "react";

const LoginForm = () => {
  const initialForm = {
    username: "",
    password: "",
  };

  const [formData, setformData] = useState(initialForm);

  const updateForm = (event) => {
    setformData({
      ...formData,
      [event.target.id]: event.target.value,
    });
  };

  const handleSubmit = (event) => {
    event.preventDefault();

    console.log(formData);
  };

  return (
    <LoginFormContainer className="login-page">
      <form className="login-form" onSubmit={handleSubmit}>
        <div className="login-form__wrapper">
          <label className="login-form__label" htmlFor="username">
            username
          </label>

          <div className="input-wrapper">
            <input
              className="login-form__input"
              id="username"
              type="username"
              autoComplete="off"
              required
              onChange={updateForm}
              value={formData.username}
            />
          </div>
        </div>

        <div className="login-form__wrapper">
          <label className="login-form__label" htmlFor="password">
            password
          </label>
          <div>
            <input
              className="login-form__input"
              id="password"
              type="password"
              autoComplete="off"
              required
              onChange={updateForm}
              value={formData.password}
            />
          </div>
        </div>

        <button className="login-form__button" type="submit">
          Log In
        </button>
      </form>

      <div className="login-form__text">
        <span>Don't have an account?</span>
      </div>

      <div>
        <button className="login-form__button" type="submit">
          Sign Up
        </button>
      </div>
    </LoginFormContainer>
  );
};

export default LoginForm;
