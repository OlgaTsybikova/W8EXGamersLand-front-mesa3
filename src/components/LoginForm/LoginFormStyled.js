import styled from "styled-components";

const LoginFormContainer = styled.div`
  background-color: white;
  border: 2px solid rgb(243, 244, 246);
  padding: 3rem 2.5rem;
  width: 448px;
  height: 512px;
  border-radius: 1.5rem;
  box-shadow: 0 20px 25px -5px rgba(0, 0, 0, 0.1),
    0 8px 10px -6px rgba(0, 0, 0, 0.1);

  .login-form {
    &__wrapper {
      padding-bottom: 1rem;
    }

    &__label {
      font-weight: 500;
      margin-bottom: 2rem;
      font-size: 1.4rem;
    }

    &__input {
      position: relative;
      width: 100%;
      border: 1px solid #222;
      border-radius: 0.375rem;
      padding: 0.75rem 0;
      outline: none;
    }

    &__button {
      margin-top: 1rem;
      width: 100%;
      padding: 0.75rem 1rem;
      font-weight: 500;
      font-size: 1rem;
      line-height: 1.25rem;
    }
  }
`;

export default LoginFormContainer;
