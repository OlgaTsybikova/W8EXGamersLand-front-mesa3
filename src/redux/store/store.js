import { configureStore } from "@reduxjs/toolkit";
import gamesReducer from "../features/gamesSlice";
import userReducer from "../features/userSlice";

const store = configureStore({
  reducer: {
    user: userReducer,
    games: gamesReducer,
  },
});

export default store;
