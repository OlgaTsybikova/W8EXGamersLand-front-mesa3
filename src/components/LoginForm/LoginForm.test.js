import { render, screen } from "@testing-library/react";

import { Provider } from "react-redux";
import store from "../../redux/store/store";
import LoginForm from "./LoginForm";

describe("Given a LoginForm component", () => {
  describe("When it's rendered", () => {
    test("Then it should render one textbox input, one password input and two buttons", () => {
      const expectedTextInputs = 1;
      const expectedPasswordInputs = 1;
      const expectedButtons = 2;

      render(
        <Provider store={store}>
          <LoginForm />
        </Provider>
      );

      const textInputs = screen.getAllByRole("textbox", { type: "text" });
      const passwordInputs = screen.getAllByLabelText("password");
      const buttons = screen.getAllByRole("button");

      expect(textInputs).toHaveLength(expectedTextInputs);
      expect(passwordInputs).toHaveLength(expectedPasswordInputs);
      expect(buttons).toHaveLength(expectedButtons);
    });
  });
});
