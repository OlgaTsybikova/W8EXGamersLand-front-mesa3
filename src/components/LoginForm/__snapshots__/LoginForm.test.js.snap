// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a LoginForm component function When invoked Then it should render correctly a form with 2 inputs and 1 buttons 1`] = `
<div
  className="sc-bczRLJ ewRgZw login-page"
>
  <form
    className="login-form"
    noValidate={true}
    onSubmit={[Function]}
  >
    <div
      className="login-form__wrapper"
    >
      <label
        className="login-form__label"
        htmlFor="username"
      >
        Username
      </label>
      <div
        className="input-wrapper"
      >
        <input
          autoComplete="off"
          className="login-form__input"
          id="username"
          onChange={[Function]}
          required={true}
          type="username"
          value=""
        />
      </div>
    </div>
    <div
      className="login-form__wrapper"
    >
      <label
        className="login-form__label"
        htmlFor="password"
      >
        Password
      </label>
      <div>
        <input
          autoComplete="off"
          className="login-form__input"
          id="password"
          onChange={[Function]}
          required={true}
          type="password"
          value=""
        />
      </div>
    </div>
    <button
      className="login-form__button"
      type="submit"
    >
      Log In
    </button>
  </form>
  <div
    className="login-form__text"
  >
    <span>
      Don't have an account?
    </span>
  </div>
  <div>
    <a
      href="/register"
      onClick={[Function]}
    >
      Register
    </a>
  </div>
</div>
`;
